torch/csrc/jit/ir.cpp

```diff
@@ -531,4 +531,29 @@ void PythonOp::cloneFrom(Node * other_) {
   }
 }
 
+void renameInputsAndOutputs(const std::shared_ptr<Graph>& graph,
+                            const std::vector<std::string>& input_names,
+                            const std::vector<std::string>& output_names) {
+  if (input_names.size() == 0 && output_names.size() == 0)
+    return;
+  if (graph->inputs().size() != input_names.size())
+    throw std::runtime_error("number of input names provided ("
+                             + std::to_string(input_names.size())
+                             + ") does not match number of inputs ("
+                             + std::to_string(graph->inputs().size())
+                             + ")");
+  if (graph->outputs().size() != output_names.size())
+    throw std::runtime_error("number of output names provided ("
+                             + std::to_string(output_names.size())
+                             + ") does not match number of outputs ("
+                             + std::to_string(graph->outputs().size())
+                             + ")");
+  for (size_t i = 0; i < input_names.size(); i++) {
+    graph->inputs()[i]->setOverridingUniqueName(input_names[i]);
+  }
+  for (size_t i = 0; i < output_names.size(); i++) {
+    graph->outputs()[i]->setOverridingUniqueName(output_names[i]);
+  }
+}
```

Comment:
Since all of this functionality is exposed in Python, wouldn't it be simpler to remove all those C++ helpers for renaming and write this code in Python as a helper to `_export`?

Reply:
modifying `overriding_unique_name_` isn't exposed in Python except through this mechanism (though of course I can expose it directly).

I wasn't aware that direct access to graph->inputs() and graph->outputs() was exposed either. If I find where that functionality is I can try moving this to python code.